dexmaker/src/test/java/com/google/dexmaker/DexMakerTest.java

```diff
@@ -1801,6 +1811,185 @@ private void addDefaultConstructor() {
         code.returnVoid();
     }
 
+    public void testCaching_Methods() throws Exception {
+        int origSize = getDataDirectory().listFiles().length;
+        final String defaultMethodName = "call";
+
+        dexMaker = new DexMaker();
+        dexMaker.declare(GENERATED, "Generated.java", PUBLIC, TypeId.OBJECT);
+        addMethodToDexMakerGenerator(TypeId.INT, defaultMethodName, TypeId.INT);
+        generateAndLoad();
+        assertEquals(origSize + 2, getDataDirectory().listFiles().length);
+
+        long lastModified  = getJarFiles()[0].lastModified();
+
+        // Create new dexmaker generator with same method signature.
+        dexMaker = new DexMaker();
+        dexMaker.declare(GENERATED, "Generated.java", PUBLIC, TypeId.OBJECT);
+        addMethodToDexMakerGenerator(TypeId.INT, defaultMethodName, TypeId.INT);
+        generateAndLoad();
+        assertEquals(origSize + 2, getDataDirectory().listFiles().length);
+        assertEquals(lastModified, getJarFiles()[0].lastModified());
+
+        // Create new dexmaker generators with different params.
+        dexMaker = new DexMaker();
+        dexMaker.declare(GENERATED, "Generated.java", PUBLIC, TypeId.OBJECT);
+        addMethodToDexMakerGenerator(TypeId.INT, defaultMethodName, TypeId.DOUBLE);
+        generateAndLoad();
+        assertEquals(origSize + 4, getDataDirectory().listFiles().length);
```

Comment:
Why is the number of files increasing by two?  You're only calling generateAndLoad() one time.